chore(docs): expand paying fees page with mana and fee analysis (#19955)
AztecProtocol/aztec-packages: docs/docs-developers/docs/foundational-topics/fees.md

This is achieved through multiple variables and calculations.

## Terminology

Familiar terms from Ethereum mainnet as referred to on the Aztec network:

| Ethereum Mainnet | Aztec              | Description                                                    |
| ---------------- | ------------------ | -------------------------------------------------------------- |

> | gas              | mana               | unit measuring computational effort for transaction operations |

Reviewer: ```suggestion
| gas              | mana               | Unit measuring computational effort for transaction operations |
```

> | fee per gas      | Fee Juice per mana | price per unit of mana                                         |

Reviewer: ```suggestion
| fee per gas      | Fee Juice per mana | Price per unit of mana                                         |
```

| fee (wei)        | Fee Juice          | total fee paid for a transaction                               |

## What is mana?

Mana is Aztec's unit of computational effort, equivalent to gas on Ethereum. Every transaction consumes mana based on the operations it performs.

Mana has two dimensions:

> - **DA (Data Availability) mana**: Cost of publishing transaction data to the data availability layer

Reviewer: ```suggestion
- **Data Availability (DA) mana**: Cost of publishing transaction data to the data availability layer
```

- **L2 mana**: Cost of executing the transaction on the Aztec network

The total transaction fee is calculated as:

```

> fee = (daGas × feePerDaGas) + (l2Gas × feePerL2Gas)

Reviewer: wait.... gas or mana ??????

Reviewer: @critesjosh still not sure if this is correct tbh, we havent defined gas on aztec

Author: ah right. will update. there are references to gas in the source code, but its not really in the docs

Author: im going to add a note about gas being used in source code

```

## What is Fee Juice?

> Fee Juice is the native fee token on Aztec, used to pay for transaction fees. It is bridged Aztec tokens from L1 and is **non-transferable** on L2 - it can only be used to pay fees, not sent between accounts.

Reviewer: ```suggestion
Fee Juice is the native fee token on Aztec, used to pay for transaction fees. It is bridged Aztec tokens from Ethereum and is **non-transferable** on Aztec - it can only be used to pay fees, not sent between accounts.
```

:::note

> Fee Juice is sometimes referred to as "Aztec token" in protocol documentation. They are the same thing.

Reviewer: is that true? i think its fee juice everywhere now isnt it

## Fee payment

> A fee payer will have Fee Juice from bridging Aztec tokens from Ethereum (L1). On Aztec this fee asset is non-transferable, and only deducted by the protocol to pay for fees. A user can claim bridged Fee Juice and use it to pay for transaction fees in the same transaction.

Reviewer: ```suggestion
A fee payer will obtain Fee Juice by bridging Aztec tokens from Ethereum (L1). On Aztec this fee asset is non-transferable, and only deducted by the protocol to pay for fees. A user can claim bridged Fee Juice and use it to pay for transaction fees in the same transaction.
```

Reviewer: potentially clearer?

> A fee payer will have Fee Juice from bridging Aztec tokens from Ethereum (L1). On Aztec this fee asset is non-transferable, and only deducted by the protocol to pay for fees. A user can claim bridged Fee Juice and use it to pay for transaction fees in the same transaction.

Reviewer: maybe say this fee payer can be the account itself or an FPC (which is like a paymaster)

> The mechanism for bridging is the same as any other token. For more on this concept see the [Token Bridge Tutorial](../tutorials/js_tutorials/token_bridge.md) which describes portal contracts and [cross-chain messaging](../aztec-nr/framework-description/how_to_communicate_cross_chain.md).

Reviewer: ```suggestion
The mechanism for bridging is the same as any other token using the native portal contracts. For more on this concept see the [Token Bridge Tutorial](../tutorials/js_tutorials/token_bridge.md) which describes portal contracts and [cross-chain messaging](../aztec-nr/framework-description/how_to_communicate_cross_chain.md).
```

Reviewer: something like this - it was confusing to me the use of the word "portal" to mean native eth <-> aztc bridge

Author: wdyt about this?  "The mechanism for bridging is the same as any other tokens, but uses enshrined native portal contracts"

### Payment methods

> An account with Fee Juice can pay for its transactions directly, including deployment of a new account, if Fee Juice has been bridged to the address where the account will be deployed.

Reviewer: im confused... does this mean you. can only deploy another account if both the og and new accoutns have fee juice? why does the new account need fee juice too ? surely thats just so the account can do stuff not so it can be deployed? This sentance is a little confusing to me

Author: im rewording it

> Alternatively, accounts can use fee-paying contracts (FPCs) to pay for transactions. FPCs accept tokens and pay fees in Fee Juice on behalf of users. Common patterns include:

Reviewer: link FPC docs
